From 57c38d1e4cbec3c4616b42cf425c4d8cb4ef5336 Mon Sep 17 00:00:00 2001
From: Nhat Thieu <nhat.thieu.xr@renesas.com>
Date: Thu, 21 Jul 2022 18:05:30 +0700
Subject: [PATCH 3/3] Add lossy compress option
Upstream-Status: Pending

Lossy compress option: suppporting customer on/off lossy conpress property.

Signed-off-by: Nhat Thieu <nhat.thieu.xr@renesas.com>
---
 meson.build          | 12 ++++++++++++
 meson_options.txt    |  2 ++
 omx/gstomxvideodec.c |  4 ++++
 3 files changed, 18 insertions(+)

diff --git a/meson.build b/meson.build
index c48c19a..c6ca601 100644
--- a/meson.build
+++ b/meson.build
@@ -170,6 +170,11 @@ else
   omx_inc = include_directories (join_paths ('omx', 'openmax'))
 endif
 
+lossy_compress = get_option('with_lossy_compress')
+if  lossy_compress == true
+   cdata.set('HAVE_LOSSY_COMPRESS', 1)
+endif
+
 default_omx_struct_packing = 0
 omx_target = get_option ('target')
 if omx_target == 'generic'
diff --git a/meson_options.txt b/meson_options.txt
index 436e4f3..ae9b567 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,5 +1,7 @@
 option('header_path', type : 'string', value : '',
     description : 'An extra include directory to find the OpenMax headers')
+option('with_lossy_compress', type : 'boolean', value : false,
+    description : 'Whether or not to support lossy compress property')
 option('target', type : 'combo',
     choices : ['none', 'generic', 'rpi', 'bellagio', 'tizonia', 'zynqultrascaleplus', 'rcar'], value : 'none',
     description : 'The OMX platform to target')
diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index c7d3283..aa44b27 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -148,9 +148,11 @@ gst_omx_video_dec_set_property (GObject * object, guint prop_id,
     case PROP_NO_REORDER:
       self->no_reorder = g_value_get_boolean (value);
       break;
+#ifdef HAVE_LOSSY_COMPRESS
     case PROP_LOSSY_COMPRESS:
       self->lossy_compress = g_value_get_boolean (value);
       break;
+#endif
     case PROP_BYPASS:
       self->bypass = g_value_get_boolean (value);
       break;
@@ -3195,7 +3197,7 @@ gst_omx_video_dec_set_format (GstVideoDecoder * decoder,
     gst_omx_component_set_parameter (self->dec, OMXR_MC_IndexParamVideoReorder,
         &sReorder);
   }
-
+#ifdef HAVE_LOSSY_COMPRESS
   if (!needs_disable) {
     /* Setting lossy compression mode (output port) */
     OMXR_MC_VIDEO_PARAM_LOSSY_COMPRESSIONTYPE sLossy;
@@ -3210,6 +3210,7 @@ gst_omx_video_dec_set_format (GstVideoDecoder * decoder,
     gst_omx_component_set_parameter (self->dec,
         OMXR_MC_IndexParamVideoLossyCompression, &sLossy);
   }
+#endif
 
   if (!needs_disable) {
     /* Setting bypass mode (output port) */
-- 
2.25.1

